### PR TITLE
add dateformat in datepicker

### PR DESCRIPTION
**Pull request by jserra91**
# PR Details
Feature: 773 task

## Description
Add our input in the datepicker and add an example in the showcase

## Related Issue
Feature: 773 task
https://github.com/systelab/systelab-components/issues/733

## Motivation and Context
Add this input for use in our project

## How Has This Been Tested
Showcase

## Types of changes
- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation 
- [x] I have updated the documentation accordingly (README.md for each UI component)
- [x] I have added tests to cover my changes (at least 1 spec for each UI component with the same coverage as the master branch)
- [x] All new and existing tests passed
- [x] A new branch needs to be created from master to evolve previous versions
- [ ] Increase version in package.json following [Semantic Versioning](https://semver.org/)
- [x] All UI components must be added into the showcase (at least 1 component with the default settings)
- [ ] Add the issue into the right [project](https://github.com/systelab/systelab-components/projects) with the proper status (In progress)

### CODE REVIEW

**Review by fjaguado — CHANGES_REQUESTED** [review]

**Review by fjaguado — APPROVED** [review]

**Review by RubKrmon — APPROVED** [review]

File: projects/systelab-components/src/lib/datepicker/README.md [hunk 1]
```diff
@@ -31,9 +31,30 @@ All the input parameters are optional.
...
+*   dd  ->  day of month (two digit)
+*   o  ->  day of the year (no leading zeros)
+*   oo  ->  day of the year (three digit)
+*   D  ->  day name short
+*   DD  ->  day name long
+*   m  ->  month of year (no leading zero)
+*   mm  ->  month of year (two digit)
+*   M  ->  month name short
+*   MM  ->  month name long
+*   y  ->  year (two digit)
```
[the remark this answers is not in the document]
Comment (OscarDeDios) [reply projects/systelab-components/src/lib/datepicker/README.md:50]: Ok, the format was already done, it's the primeng format

**Comment by OscarDeDios** [comment]
Don't forget increase version in package.json in order to publish a new version

File: projects/systelab-components/src/lib/datepicker/datepicker.component.ts [hunk 2]
```diff
@@ -365,9 +366,10 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
 		};
 
 		this.language.firstDayOfWeek = this.i18nService.getFirstDayOfWeek();
-		this.language.dateFormatValue = this.i18nService.getDateFormatForDatePicker(true);
+
+		this.language.dateFormatValue = this.dateFormat ? this.dateFormat : this.i18nService.getDateFormatForDatePicker(true);
 		if (this.currentCalendar) {
-			this.currentCalendar.dateFormat = this.language.dateFormatValue;
+			this.currentCalendar.dateFormat = !this.dateFormat ? this.language.dateFormatValue : this.dateFormat;
```
[the remark this answers is not in the document]
Comment (amgart) [reply projects/systelab-components/src/lib/datepicker/datepicker.component.ts:372]: I am not sure to understand this change... 
In case that this.dateFormat is present in line 370, this.language.dateFormat has same value that this.dateFormat... so this statement is irrelevant. You could just use this.language.dateFormatValue as before.
In case that this.dateFormat is not present in line 370, this.language.dateFormat has the value 'this.i18Service.getDateFormatForDatePicker(true)'... so line 372 will always set the first statement.

In summary... I think this change is not needed but maybe I am missing something... ¿?

**Comment by amgart** [comment]
No package.json version update is needed?

**Review by alfonsserra — APPROVED** [review]
Thank you so much @jserra91

**Review by RubKrmon — APPROVED** [review]

**Review by oGalceran1 — APPROVED** [review]

**Review by OscarDeDios — APPROVED** [review]
